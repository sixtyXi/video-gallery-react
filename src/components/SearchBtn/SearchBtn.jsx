import React from 'react';
import PropTypes from 'prop-types';

import styles from './SearchBtn.scss';

export const SearchBtn = ({ className }) => {
  return (
    <button className={`${styles.searchBtn} ${styles[className]}`} type="button">
      Search
    </button>
  );
};

SearchBtn.propTypes = {
  className: PropTypes.oneOf(['searchBtnPrimary', 'searchBtnSecondary'])
};

SearchBtn.defaultProps = {
  className: 'searchBtnPrimary'
};
