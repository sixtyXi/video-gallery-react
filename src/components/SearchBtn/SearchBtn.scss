@import '../../shared/_colors.scss';
@import '../../shared/_typography.scss';

.searchBtn {
  padding: 5px 40px;
  font-variant: all-small-caps;
  border: none;
  cursor: pointer;
}

.searchBtnPrimary {
  background: $accent;
  color: $text-light;
}

.searchBtnSecondary {
  padding: 5px 20px;
  background: white;
  color: $accent;
}
