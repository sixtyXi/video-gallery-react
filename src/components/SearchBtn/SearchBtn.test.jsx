import React from 'react';
import { shallow } from 'enzyme';

import { SearchBtn } from './SearchBtn';

describe('SearchBtn', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<SearchBtn />);

    expect(wrapper).toMatchSnapshot();
  });
});
