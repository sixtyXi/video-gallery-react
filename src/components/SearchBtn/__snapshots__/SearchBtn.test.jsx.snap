// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBtn renders correctly 1`] = `
<button
  className="searchBtn searchBtnPrimary"
  type="button"
>
  Search
</button>
`;
